provisioning_templates/provision/kickstart_default.erb

```diff
@@ -94,14 +84,81 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 skipx
 
-<% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
-<% else -%>
-<% dhcp = !@static -%>
-<% end -%>
+<%
+  network_options = []
+  nameservers = []
+  subnet4 = iface.subnet
+  subnet6 = iface.subnet6
+
+  # device and hostname
+  network_options.push("--device=#{iface.mac || iface.identifier}")
+  network_options.push("--hostname #{@host.name}")
+
+  # single stack
+  if subnet4 && !subnet6
+    network_options.push("--noipv6")
+  elsif !subnet4 && subnet6
+    network_options.push("--noipv4")
+  end
+
+  # dual stack MTU check
+  raise("IPv4 and IPv6 subnets have different MTU") if subnet4 && subnet6 && subnet4.mtu.present? && subnet6.mtu.present? && subnet4.mtu != subnet6.mtu
+
+  # IPv4
+  if (subnet4 && !subnet4.dhcp_boot_mode?) || @static
+    network_options.push("--bootproto static")
+    network_options.push("--ip=#{iface.ip}")
+    network_options.push("--netmask=#{subnet4.mask}")
+    network_options.push("--gateway=#{subnet4.gateway}")
+  elsif subnet4 && subnet4.dhcp_boot_mode?
+    network_options.push("--bootproto dhcp")
+  end
+  if subnet4
+    nameservers.push(subnet4.dns_servers)
+    network_options.push("--mtu=#{subnet4.mtu}") if subnet4.mtu.present?
+  end
 
-network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu.to_s}" %> --hostname <%= @host %> <%= network_options.join(' ') %>
+  # IPv6
+  if rhel_compatible && os_major >= 6
+    if (subnet6 && !subnet6.dhcp_boot_mode?) || @static6
+      network_options.push("--ipv6=#{iface.ip6}/#{subnet6.cidr}")
+      network_options.push("--ipv6gateway=#{subnet6.gateway}")
+    elsif subnet6 && subnet6.dhcp_boot_mode?
+      if host_param_true?('use-slaac')
+        network_options.push("--ipv6 auto")
+      else
+        network_options.push("--ipv6 dhcp")
+      end
+    end
+    if subnet6
+      nameservers.push(subnet6.dns_servers)
+      network_options.push("--mtu=#{subnet6.mtu}") if subnet6.mtu.present?
+    end
+  end
+
+  # bond
+  if iface.bond? && rhel_compatible && os_major >= 6
+    network_options.push("--bondslaves=#{iface.attached_devices_identifiers}")
+    network_options.push("--bondopts=mode=#{iface.mode};#{iface.bond_options.tr(' ', ';')}")
+  end
+
+  # VLAN (only on physical is recognized)
+  if iface.virtual? && iface.tag.present? && iface.attached_to.present?
+    if rhel_compatible && os_major >= 6
+      network_options.push("--vlanid=#{iface.tag}")
+    elsif rhel_compatible && os_major >= 7
+      network_options.push("--interfacename=vlan#{iface.tag}")
+    end
+  end
+
+  # DNS
+  if nameservers.flatten.size > 0
```

Comment:
Flatten would not be needed if `+=` was used earlier

Reply:
Oh I can use `concat` or `push(*array)`. Will fix.